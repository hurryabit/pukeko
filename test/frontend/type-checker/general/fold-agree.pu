data List a =
  | Nil
  | Cons a (List a)
fold : (b -> a -> b) -> b -> List a -> b
fold f y0 xs =
  match xs with
  | Nil -> y0
  | Cons x xs -> fold f (f y0 x) xs
