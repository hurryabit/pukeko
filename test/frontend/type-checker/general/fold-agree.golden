data List a =
       | Nil
       | Cons a (List a)
fold : forall a b. (b -> a -> b) -> b -> List a -> b =
  fun @a @b ->
    fun (f : b -> a -> b) ->
      fun (y0 : b) ->
        fun (xs : List a) ->
          match xs with
          | Nil -> y0
          | Cons x xs -> fold @a @b f (f y0 x) xs
